2.train_model/scripts/nbconverted/train_single_class_models.py

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[1]:
+
+
+import pathlib
+import warnings
+
+import pandas as pd
+import numpy as np
+
+from sklearn.linear_model import LogisticRegression
+from sklearn.model_selection import (
+    StratifiedKFold,
+    GridSearchCV,
+)
+from sklearn.utils import shuffle, parallel_backend
+from sklearn.exceptions import ConvergenceWarning
+from joblib import dump
+
+import sys
+
+sys.path.append("../utils")
+from split_utils import get_features_data
+from train_utils import get_dataset, get_X_y_data
+
+
+# In[2]:
+
+
+# set numpy seed to make random operations reproduceable
+np.random.seed(0)
+
+# load training data from indexes and features dataframe
+data_split_path = pathlib.Path(f"../1.split_data/indexes/data_split_indexes.tsv")
+features_dataframe_path = pathlib.Path("../0.download_data/data/labeled_data.csv.gz")
+
+# dataframe with only the labeled data we want (exclude certain phenotypic classes)
+features_dataframe = get_features_data(features_dataframe_path)
+data_split_indexes = pd.read_csv(data_split_path, sep="\t", index_col=0)
+
+# get training data from labeled data
+training_data = get_dataset(features_dataframe, data_split_indexes, "train")
+# training_data
+
+
+# In[3]:
+
+
+# specify model types and feature types
+model_types = ["final", "shuffled_baseline"]
+feature_types = ["CP", "DP", "CP_and_DP"]
```

Comment:
It seems that you are creating combinations with these two lists. Consider using `itertools.combinations` 

There are other types of combinations [algorithms](https://docs.python.org/3/library/itertools.html#module-itertools) select that one that best suits for you.

Reply:
I ended up using `itertools.product`. Thanks for the suggestion!